src/Invoice_GenUI/Invoice_GenUI/ViewModels/InvoiceDetailsViewModel.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using Invoice_GenUI.Models;
+using Invoice_GenUI.Models.Services;
+
+namespace Invoice_GenUI.ViewModels
+{
+    public partial class InvoiceDetailsViewModel : ViewModel
+    {
+        [ObservableProperty]
+        private INavigationService _navigation;
+        private readonly IInvoiceService _invoiceService;
+        private readonly IClientService _clientService;
+        private readonly ShowInvoicesViewModel _showInvoicesViewModel;
+
+        public ObservableCollection<LineItemModel> LineItemDetails { get; set; }
+
+        public InvoiceDetailsViewModel(INavigationService navService, IInvoiceService invoiceService, ShowInvoicesViewModel showInvoicesViewModel, IClientService clientService)
+        {
+            _navigation = navService;
+            _invoiceService = invoiceService;
+            _showInvoicesViewModel = showInvoicesViewModel;
+            _clientService = clientService;
+            Task.Run(() => GetInvoiceID()).Wait();
+            AssignTotal();
+        }
+        public string? ClientName { get; set; }
+        public double VatRate { get; set; }
+        public double Total { get; set; }
+        public double InvoiceTotal { get; set; }
+        public DateTime IssueDate { get; set; }
+        public DateTime DueDate { get; set; }
+        private async Task GetInvoiceID()
+        {
+            int ID;
+            var singleInvoice = await _invoiceService.GetSingleInvoiceDetails(_showInvoicesViewModel.InvoiceID);
+
+            VatRate = singleInvoice.VatRate;
+            IssueDate = singleInvoice.IssueDate.Date;
+            DueDate = singleInvoice.DueDate.Date;
+            LineItemDetails = singleInvoice.LineItems;
+
+            foreach (var item in singleInvoice.LineItems)
+            {
+                Total += item.Cost * item.Quantity;
+                var vatTotal = Total * (singleInvoice.VatRate / 100);
+                InvoiceTotal = Total + vatTotal;
+            }
+
+            ID = singleInvoice.ClientId;
```

Comment:
🔧 It would make a little more sense to declare `id` here rather than on line 38. Also please use lower camel case for variable names.